capsules/src/usb/cdc.rs

```diff
@@ -0,0 +1,496 @@
+    /// can pass it back to the client when the transmission has finished.
+    tx_len: Cell<usize>,
+    /// Where in the `tx_buffer` we need to start sending from when we continue.
+    tx_offset: Cell<usize>,
+    /// The TX client to use when transmissions finish.
+    tx_client: OptionalCell<&'a dyn uart::TransmitClient>,
+
+    /// A holder for the buffer to receive bytes into. We use this as a flag as
+    /// well, if we have a buffer then we are actively doing a receive.
+    rx_buffer: TakeCell<'static, [u8]>,
+    /// How many bytes the client wants us to receive.
+    rx_len: Cell<usize>,
+    /// How many bytes we have received so far.
+    rx_offset: Cell<usize>,
+    /// The RX client to use when RX data is received.
+    rx_client: OptionalCell<&'a dyn uart::ReceiveClient>,
+}
+
+impl<'a, U: hil::usb::UsbController<'a>> CdcAcm<'a, U> {
+    pub fn new(controller: &'a U, max_ctrl_packet_size: u8) -> Self {
+        let interfaces: &mut [InterfaceDescriptor] = &mut [
+            InterfaceDescriptor {
+                interface_number: 0,
+                interface_class: 0x02,    // CDC communication
+                interface_subclass: 0x02, // abstract control model (ACM)
+                interface_protocol: 0x01, // V.25ter (AT commands)
+                ..InterfaceDescriptor::default()
+            },
+            InterfaceDescriptor {
+                interface_number: 1,
+                interface_class: 0x0a,    // CDC data
+                interface_subclass: 0x00, // none
+                interface_protocol: 0x00, // none
+                ..InterfaceDescriptor::default()
+            },
+        ];
+
+        let cdc_descriptors: &mut [CdcInterfaceDescriptor] = &mut [
+            CdcInterfaceDescriptor {
+                subtype: descriptors::CdcInterfaceDescriptorSubType::Header,
+                field1: 0x10, // CDC
+                field2: 0x11, // CDC
+            },
+            CdcInterfaceDescriptor {
+                subtype: descriptors::CdcInterfaceDescriptorSubType::CallManagement,
+                field1: 0x00, // Capabilities
+                field2: 0x01, // Data interface 1
+            },
+            CdcInterfaceDescriptor {
+                subtype: descriptors::CdcInterfaceDescriptorSubType::AbstractControlManagement,
+                field1: 0x06, // Capabilities
+                field2: 0x00, // unused
+            },
+            CdcInterfaceDescriptor {
+                subtype: descriptors::CdcInterfaceDescriptorSubType::Union,
+                field1: 0x00, // Interface 0
+                field2: 0x01, // Interface 1
+            },
+        ];
+
+        let endpoints: &[&[EndpointDescriptor]] = &[
+            &[EndpointDescriptor {
+                endpoint_address: EndpointAddress::new_const(4, TransferDirection::DeviceToHost),
+                transfer_type: TransferType::Interrupt,
+                max_packet_size: 8,
+                interval: 100,
```

Comment:
I don't think this value actually matters very much since we don't do anything with the interrupt endpoint, but TinyUSB sets their interval to 16 and copying them seems reasonable. https://github.com/hathach/tinyusb/blob/e9b41cb97d17e82287a146aba1159897a3bcfed2/src/device/usbd.h#L211

```suggestion
                interval: 16,
```